### PR TITLE
added functional tests for the charm.

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- removed installation as classic
- adapted unit test because of changes
- updated `.gitignore` according with bootstack templates.
- updated the installation method of the snap to be able to fecth from snapstore

### CODE REVIEW

**Review by esunar — COMMENTED** [review]
recheck

**Review by chanchiwai-ray — DISMISSED** [review]
LGTM, (non-blocking) it seems that there are 3 independent PRs packed in this PR, it would be nicer to have 3 small PRs (or at least 3 commits).

- apply bootstack template (`.gitignore`, `tox.ini`)
- add basic functional tests (`tests/*`)
- fix unit tests (`src/charm.py`, `unit/test_charm.py`)

**Review by esunar — APPROVED** [review]

File: src/charm.py
Comment (mkalcok) [inline src/charm.py:81]: Why was this change needed?

File: tests/functional/tests/bundles/resources/software-inventory-exporter.snap
Comment (mkalcok) [inline tests/functional/tests/bundles/resources/software-inventory-exporter.snap:1]: It's not a good idea to merge 31MB of binary data to the repository as it's gonna stay there forever. These functional tests will have to wait until the snap is published in the store before we can merge it.

File: tests/functional/tests/bundles/overlays/local-charm-overlay.yaml.j2
Comment (mkalcok) [inline tests/functional/tests/bundles/overlays/local-charm-overlay.yaml.j2:6]: It's fine to use local resource for development of tests before the snap is in the store, but it can't be pushed to the main repo and used in CI.

File: tests/functional/tests/test_charm.py
Comment (mkalcok) [inline tests/functional/tests/test_charm.py:38]: If it's not overly complicated, I think it'd be better to also test that the endpoint provide correct data (e.g. all snaps/kernels/packages are reported by the exporter)

**Review by mkalcok — CHANGES_REQUESTED** [review]
Overall this change is fine, but the inclusion of snap as a local resource is a blocker. We have to wait until it's published in the snapstore.

File: tests/functional/requirements.txt
Comment (rgildein) [inline tests/functional/requirements.txt:2]: Can you remove unnecessary zaza requirements here? Thanks

File: tests/functional/tests/bundles/resources/software-inventory-exporter.snap
Comment (rgildein) [reply tests/functional/tests/bundles/resources/software-inventory-exporter.snap:1]: I agree w/ Martin here, we can pull code a build snap if that is really needed, but it's ok to wait until snap is not released.

**Review by esunar — CHANGES_REQUESTED** [review]

**Review by rgildein — DISMISSED** [review]
The changes look good to me, but I think this is a good example of a PR that should be split into two PRs. Can you please update description in PR, thanks.

**Review by esunar — DISMISSED** [review]
Approved with a nice to have inline comment.

**Review by rgildein — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Review by esunar — APPROVED** [review]

**Pull request merged**
